File: src/main/kotlin/com/tarehart/draw/RainbowVendor.kt
package com.tarehart.draw

import java.awt.Color
import java.util.concurrent.atomic.AtomicInteger

class RainbowVendor(private val segments: Int) {
    private val index = AtomicInteger(0)

    fun nextColor(): Color {
        val hue = (index.getAndIncrement() % segments) / segments.toFloat() // Normalize to [0, 1]
        return Color.getHSBColor(hue, 1.0f, 1.0f) // Full saturation and brightness
    }
}

File: draw/rainbow_vendor.py
from colorsys import hsv_to_rgb
from typing import Tuple


class RainbowVendor:
    def __init__(self, segments: int):
        self.index = 0
        self.segments = segments

    def next_color(self) -> Tuple[int, int, int]:
        hue = (self.index % self.segments) / self.segments
        rgb = hsv_to_rgb(hue, 1, 1)
        self.index += 1
        return int(rgb[0] * 255), int(rgb[1] * 255), int(rgb[2] * 255)